src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -523,44 +523,54 @@ EnsureSequenceTypeSupported(Oid relationId, AttrNumber attnum, Oid seqTypId)
 							" is not supported for distribution")));
 	}
 
-	Oid seqOid = lfirst_oid(list_head(sequencesFromAttrDef));
-
-	List *citusTableIdList = CitusTableTypeIdList(ANY_CITUS_TABLE_TYPE);
-	Oid citusTableId = InvalidOid;
-	foreach_oid(citusTableId, citusTableIdList)
+	/*
+	 * We need this check because sometimes there are cases where the
+	 * dependency between the table and the sequence is not formed
+	 * One example is when the default is defined by
+	 * DEFAULT nextval('seq_name'::text) (not by DEFAULT nextval('seq_name'))
+	 * In these cases, sequencesFromAttrDef with be empty.
+	 */
+	else if (list_length(sequencesFromAttrDef) == 1)
```

Comment:
I was a slightly confused by this check because it is checking the opposite of what the comment describes.

If could be nice to have something like
```
if (list_length(sequencesFromAttrDef) == 0)
{
    /* ... describe this case ... */
    return;
}

if (list_length(sequencesFromAttrDef) > 1)
{
    /* we do not support cases like "DEFAULT nextval('..') - nextval('..')" */
    ereport(ERROR, ...);
}

```